internal/handlers/common/distinct.go

```diff
@@ -102,26 +102,18 @@ func FilterDistinctValues(iter types.DocumentsIterator, key string) (*types.Arra
 			return nil, lazyerrors.Error(err)
 		}
 
-		// docsAtSuffix contains all documents exist at the suffix key.
-		docsAtSuffix := []*types.Document{doc}
-		suffix := key
-
-		if strings.ContainsRune(key, '.') {
-			path, err := types.NewPathFromString(key)
-			if err != nil {
-				return nil, lazyerrors.Error(err)
-			}
-
-			// Multiple documents may be found at suffix by array dot notation.
-			suffix, docsAtSuffix = getDocumentsAtSuffix(doc, path)
+		path, err := types.NewPathFromString(key)
+		if err != nil {
+			return nil, lazyerrors.Error(err)
 		}
 
-		for _, doc := range docsAtSuffix {
-			val, err := doc.Get(suffix)
-			if err != nil {
-				continue
-			}
+		// vals contains all values exist at the suffix of the path
+		vals, err := commonpath.FindValues(doc, path, commonpath.FindValuesOpts{})
```

Comment:
Small suggestion: I'd suggest to allow using nil for opts that will set default values (at least that's what I think we do in most of options in the code)
```suggestion
		vals, err := commonpath.FindValues(doc, path, nil)
```